lib/hash.c

```diff
@@ -0,0 +1,30 @@
+#include "oomph.h"
+#include <stdio.h>
+#include <openssl/evp.h>
+
+// TODO: accept arbitrary binary data instead of utf-8 strings
+struct class_Str *oomph_hash(const struct class_Str *data, const struct class_Str *algname)
+{
+	const EVP_MD *alg = EVP_get_digestbyname(algname->str);
+	if (!alg)
+		panic_printf("unknown hash algorithm name: %s", algname->str);
```

Comment:
need test

Reply:
actually not, because for this to fail, some function in stdlib/hash.oomph would have to fail